Frameworks/include/ABMultiValueInternal.h

```diff
@@ -0,0 +1,44 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#import <AddressBook/ABMultiValue.h>
+#import <Foundation/Foundation.h>
+
+@interface _ABMultiValue : NSObject
+
+- (id)initWithPropertyType:(ABPropertyType)propertyType;
+
+// Once the MultiValue is made immutable,
+// appendPairWithLabel:andValue: no longer does anything;
+// it returns YES if the appending occurred, or NO if the
+// MultiValue is immutable and no change was made.
```

Comment:
There is also ABMutableMultiValueRef (the idea being that the mutable version simply doesn't mark itself as immutable so that new pairs could be added/existing ones changed)

---
In reply to: [75180337](https://github.com/Microsoft/WinObjC/pull/884#discussion_r75180337) [](ancestors = 75180337)